src/terminal.ts

```diff
@@ -0,0 +1,354 @@
+  debug: Debug
+): vscode.Terminal {
+  const pty = new DecoderTerminal(arduinoContext, decoder, debug);
+  const options: vscode.ExtensionTerminalOptions = {
+    name: decodeTerminalName,
+    pty,
+    iconPath: new vscode.ThemeIcon('debug-console'),
+  };
+  return vscode.window.createTerminal(options);
+}
+
+const decodeTerminalTitle = 'ESP Exception Decoder';
+const decodeTerminalName = 'Exception Decoder';
+const initializing = 'Initializing...';
+const busy = 'Decoding...';
+const idle = 'Paste exception to decode...';
+
+interface DecodeTerminalState {
+  params: DecodeParams | Error;
+  userInput?: string | undefined;
+  decoderResult?: DecodeResult | Error | undefined;
+  statusMessage?: string | undefined;
+}
+
+class DecoderTerminal implements vscode.Pseudoterminal {
+  readonly onDidWrite: vscode.Event<string>;
+  readonly onDidClose: vscode.Event<number | void>;
+
+  private readonly onDidWriteEmitter: vscode.EventEmitter<string>;
+  private readonly onDidCloseEmitter: vscode.EventEmitter<number | void>;
+  private readonly toDispose: vscode.Disposable[];
+
+  private state: DecodeTerminalState;
+  private abortController: AbortController | undefined;
+
+  constructor(
+    private readonly arduinoContext: ArduinoContext,
+    private readonly decoder: typeof decode = decode,
+    private readonly debug: Debug = terminalDebug
+  ) {
+    this.onDidWriteEmitter = new vscode.EventEmitter<string>();
+    this.onDidCloseEmitter = new vscode.EventEmitter<number | void>();
+    this.toDispose = [
+      this.onDidWriteEmitter,
+      this.onDidCloseEmitter,
+      this.arduinoContext.onDidChange('boardDetails')(() =>
+        this.updateParams()
+      ), // ignore FQBN changes and listen on board details only
+      this.arduinoContext.onDidChange('compileSummary')(() =>
+        this.updateParams()
+      ),
+      this.arduinoContext.onDidChange('sketchPath')(() => this.updateParams()), // In the Arduino IDE (2.x), the sketch path does not change
+      new vscode.Disposable(() => this.abortController?.abort()),
+    ];
+    this.onDidWrite = this.onDidWriteEmitter.event;
+    this.onDidClose = this.onDidCloseEmitter.event;
+    this.state = {
+      params: new Error(initializing),
+      statusMessage: idle,
+    };
+  }
+
+  open(): void {
+    this.updateParams();
+  }
+
+  close(): void {
+    vscode.Disposable.from(...this.toDispose).dispose();
+  }
+
+  handleInput(data: string): void {
+    this.debug(`handleInput: ${data}`);
+    if (data.trim().length < 2) {
+      this.debug(`handleInput, skip: ${data}`);
+      // ignore single keystrokes
+      return;
+    }
+    if (this.state.params instanceof Error) {
+      this.debug(`handleInput, skip: ${this.state.params.message}, ${data}`);
+      // ignore any user input if the params is invalid
+      return;
+    }
+    const params = this.state.params;
+    this.updateState({
+      userInput: toTerminalEOL(data),
+      statusMessage: busy,
+    });
```

Comment:
should update the `decoderResult` to `undefined`.

Reply:
https://github.com/dankeboy36/esp-exception-decoder/pull/1/commits/eb1fb4ff969de8e7f47c2744d8d91bc3faa83ea7